src/HDF5.jl

```diff
@@ -535,25 +536,33 @@ heuristic_chunk(x) = Int[]
 
 ### High-level interface ###
 # Open or create an HDF5 file
-function h5open(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Bool, ff::Bool,
+function h5open(filename::AbstractString, mode::String,
```

Comment:
I think you're right that the signature could be AbstractString. I think the docstring should be in only one place. I put it on `function h5open end` because I thought that was the preferred way to document a function as a whole rather than just a method, but the output of `?h5open` is the same if I just move the docstring here.